### PR TITLE
Reduce url re-encoding

### PR DESCRIPTION
**Pull request by bdraco**

### CODE REVIEW

File: src/synology_dsm/synology_dsm.py [hunk 1]
```diff
@@ -110,9 +110,9 @@ def __init__(
 
         # Build variables
         if use_https:
-            self._base_url = f"https://{dsm_ip}:{dsm_port}"
+            self._base_url = URL(f"https://{dsm_ip}:{dsm_port}")
```
Comment (bdraco) [inline src/synology_dsm/synology_dsm.py:113]: By building this here we avoid calling `[_make_netloc](https://github.com/aio-libs/yarl/blob/bd5ff24f8a5a3347bf3251237b6d32c40e21b928/yarl/_url.py#L197)` on every request which has to make an ip_address object and than convert it back to a string

**Comment by bdraco** [comment]
This works fine on my system, but I don't really have a good way to test the encoding is preserved like #261

**Comment by mib1185** [comment]
> This works fine on my system, but I don't really have a good way to test the encoding is preserved like #261

it was initially added with #260 to fix https://github.com/home-assistant/core/issues/87457
so an easy way to test would be using a username containing spaces

**Comment by bdraco** [comment]
```python
import yarl
from urllib.parse import urlencode, quote

params = {'user': 'with spaces'}
query = urlencode(params, safe="?/:@-._~!$'()*,", quote_via=quote)
url = yarl.URL('http://example.com')
url_encoded = url.join(yarl.URL(f"?{query}", encoded=True))
print(url_encoded)

print (url_encoded.query)
```

```
http://example.com/?user=with%20spaces
<MultiDictProxy('user': 'with spaces')>
```

**Comment by bdraco** [comment]
behavior is the same as
```
import yarl
from urllib.parse import urlencode, quote

params = {'user': 'with spaces'}
safe = "?/:@-._~!$'()*,"
url = 'http://example.com'
query = urlencode(params, safe=safe, quote_via=quote)
url_encoded = yarl.URL(str(yarl.URL(url)) + "?" + query, encoded=True)
print(url_encoded)

print (url_encoded.query)
```

**Review by mib1185 — APPROVED** [review]
LGTM, thanks @bdraco 👍

**Pull request merged**